Develop

NastyaPilat opened the pull request.

Commit:
new file:   conftest.py
	new file:   tests/helpers.py
	new file:   tests/test_login.py
	new file:   tests/test_registration.py

Commit:
modified:   README.md
	new file:   __pycache__/conftest.cpython-312-pytest-8.1.1.pyc
	new file:   requirements.txt
	new file:   tests/__pycache__/helpers.cpython-312.pyc
	new file:   tests/__pycache__/test_ingredients_tabs.cpython-312-pytest-8.1.1.pyc
	new file:   tests/__pycache__/test_login.cpython-312-pytest-8.1.1.pyc
	new file:   tests/__pycache__/test_logout.cpython-312-pytest-8.1.1.pyc
	new file:   tests/__pycache__/test_registration.cpython-312-pytest-8.1.1.pyc
	modified:   tests/helpers.py
	new file:   tests/test_ingredients_tabs.py
	modified:   tests/test_login.py
	new file:   tests/test_logout.py
	modified:   tests/test_registration.py

irisqul commented on __pycache__/conftest.cpython-312-pytest-8.1.1.pyc:
Нужно исправить: из финального проекта нужно удалить все файлы и папки, которые не относятся к решению задачи

irisqul commented on conftest.py:
```diff
@@ -0,0 +1,17 @@
+
+@pytest.fixture
+def base_url():
+    return "https://stellarburgers.nomoreparties.site/"
```
Нужно исправить: нам не нужны фикстуры чтобы получить статичные данные, нужно перенести в константы и просто получать по имени, а не вызывать функции. В фикстурах можно, например, генерировать новые данные для тестирования, уникальные для каждого теста.

irisqul commented on conftest.py:
```diff
@@ -0,0 +1,17 @@
+
+@pytest.fixture
+def credentials():
+    return { 'name': 'test_user', 'email': 'test@domain.ru', 'password': 'test_password' }
```
Нужно исправить: нам не нужны фикстуры чтобы получить статичные данные

irisqul commented on tests/__pycache__/helpers.cpython-312.pyc:
Нужно исправить: удалить все лишное файлы

irisqul commented on tests/helpers.py:
```diff
@@ -0,0 +1,41 @@
+    
+
+def perform_registration(driver, name, email, password):
+    [input_name, input_email, input_password] = driver.find_elements(By.TAG_NAME, "input")
```
Нужно исправить: В этом проекте пока не проходим создание методов для взаимодействия с приложением, в следующем спринте будет объеснено какую тестовую архитектуру нужно делать.. Пока не получилось сделать эффективно, пожалуйста давай все шаги делать внутри тестов.

irisqul commented on tests/helpers.py:
```diff
@@ -0,0 +1,41 @@
+    input_email.send_keys(email)
+    input_password.send_keys(password)
+    
+    registration_button = driver.find_element(By.XPATH, "//button[text() = 'Зарегистрироваться']")
```
Нужно исправить: то что проходим в этом спринте - это локаторы, нужно сделать файл в котором константами записать все локаторы, и далее везде вызывать локаторы по именам.

irisqul commented on tests/test_ingredients_tabs.py:
```diff
@@ -0,0 +1,33 @@
+
+def test_bun_tab(driver, base_url):
+    driver.get(base_url)
+
```
Можно лучше: Здесь и далее. Проследи за соблюдением стиля PEP8 в твоем коде, внутри функций - отступы не делаются, между методами одного класса - отступ в 1 строку

irisqul commented on tests/test_ingredients_tabs.py:
```diff
@@ -0,0 +1,33 @@
+    driver.get(base_url)
+
+    bun_tab = driver.find_element(By.XPATH, "//div[span[text() = 'Булки']]")
+    sauce_tab = driver.find_element(By.XPATH, "//div[span[text() = 'Соусы']]")
```
Нужно исправить: локаторы внутри тестов уже не упоминаем, нужно сделать отдельный файл с локаторами

irisqul commented on tests/test_ingredients_tabs.py:
```diff
@@ -0,0 +1,33 @@
+    bun_tab = driver.find_element(By.XPATH, "//div[span[text() = 'Булки']]")
+    sauce_tab = driver.find_element(By.XPATH, "//div[span[text() = 'Соусы']]")
+
+    sauce_tab.click()
```
Можно просто кликнуть по элементу, а не отдельно искать и кликать

irisqul commented on tests/test_login.py:
```diff
@@ -0,0 +1,59 @@
+    login_button.click()
+
+    if not helpers.wait_for_url_contains(driver, 'login'):
+        pytest.fail("URL не изменился в течение 3 секунд до содержимого 'login'.")
```
Нужно исправить: тестирования помимо ассертов в тестах быть не должно. Мы хотим написать линейный сценарий и проверить только на последнем этапе результат.

irisqul commented on tests/test_registration.py:
```diff
@@ -0,0 +1,25 @@
+def test_successful_registration(driver, base_url, credentials):
+    driver.get(base_url + 'register')
+
+    email = ''.join(random.choices(string.ascii_letters, k=6)) + '@domain.ru'
```
Можно лучше: генерацию данных для тестовой сессии можно перенести в фикстуру

Commit:
new file:   .gitignore
	deleted:    __pycache__/conftest.cpython-312-pytest-8.1.1.pyc
	modified:   conftest.py
	new file:   constants.py
	new file:   locators.py
	deleted:    tests/__pycache__/helpers.cpython-312.pyc
	deleted:    tests/__pycache__/test_ingredients_tabs.cpython-312-pytest-8.1.1.pyc
	deleted:    tests/__pycache__/test_login.cpython-312-pytest-8.1.1.pyc
	deleted:    tests/__pycache__/test_logout.cpython-312-pytest-8.1.1.pyc
	deleted:    tests/__pycache__/test_registration.cpython-312-pytest-8.1.1.pyc
	deleted:    tests/helpers.py
	modified:   tests/test_ingredients_tabs.py
	modified:   tests/test_login.py
	modified:   tests/test_logout.py
	modified:   tests/test_registration.py

irisqul commented on constants.py:
```diff
@@ -0,0 +1,2 @@
+BASE_URL = "https://stellarburgers.nomoreparties.site/"
+CREDENTIALS = { 'name': 'test_user', 'email': 'test@domain.ru', 'password': 'test_password' }
```
Можно лучше: урлы и тестовые данные - это очень разная информация, стоит разделять по классам
